### PR TITLE
Minimum docs for bugbash

### PR DESCRIPTION
**Pull request by clearab**
This is the minimum docs necessary for the bug bash. Covers:

- Edits to base readme
- Remove readme files from /packages and /templates folders (contents moved to base readme)
- overview, extending with code, and extending with packages documents
- stubs for package creation and template creation documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,79 +15,52 @@ This repository contains the source code for *components* published by Microsoft
...
-
-For example, the Conversational Core template will take a dependency on two packages - welcome and  help & cancel. It will also include a root dialog that wires up the dialogs in those packages as well as a dialog for handling unkown intents. This represents the base set of functionality nearly all conversational bots include. If you were to start from the empty/echo bot template, you could choose to add these packages later - either way you'd get the same set of functionality (without the need to do something like compare code samples and try and stitch them together yourself).
-
-## Skills
+## Documentation
 
-Skills are separate bots you connect your bot to in order to process messages for you. The skill manifest establishes a contract other bots can follow - defining messages and events the skill can handle and any data that will be returned when the skill completes its interaction with your user.
+We document working with components [here][./docs/overview.md], and you can find the full documentation for the Bot Framework SDK & Composer [here](https://aka.ms/botdocs).
```
Comment (benbrown) [inline README.md:20]: bad link format in the [here] link

File: README.md [hunk 2]
```diff
@@ -16,79 +15,52 @@ This repository contains the source code for *components* published by Microsoft
...
 ### Templates
 
-| Name         | Description |  C#  | JavaScript |
-|:------------:|-------------|------|------------|
-|Echo Bot | The base empty bot | `someplace` | `someplace` |
+Our [yeoman](https://yeoman.io) generators for scaffolding bot projects.
+
+| Name         | Description |
+|:------------:|-------------|
+|[Empty bot](~/generators/generator-microsoft-bot-empty) ) | The base empty bot |
```
Comment (benbrown) [inline README.md:30]: all these links with ~s in them are broken

File: docs/overview.md [hunk 3]
```diff
@@ -1,4 +1,31 @@
 # The Component Model
 
-asdf
+The Component Model for bots is our framework for sharing and reusing bot functionality - through new bots with templates, using portions of bots in the form of packages, or as complete bots with skills.
 
+To work with components from Composer while this feature is in preview, you'll need to enable this feature in your Composer settings menu - Settings > Preview Features > New Creation Experience.
+
+## Table of Contents
+
+1. [Overview](~/docs/overview.md)
```
Comment (peterinnesmsft) [inline docs/overview.md]: Pointing to itself. Remove?

File: docs/overview.md [hunk 4]
```diff
@@ -1,4 +1,31 @@
...
+4. [Creating your own packages](~/docs/creating-packages.md)
+5. [Creating your own templates](~/docs/creating-templates.md)
+
+## Adaptive Runtime
+
+At the core of the component model is the adaptive runtime - an extensible, configurable runtime that is treated as a black box to bot developers and taken as a dependency. The runtime provides extensibility points to add additional functionality by importing packages, connection to skills, or adding your own coded extensions.
+
+## Packages
+
+Packages are bits of a bot you want to share/import like declarative dialog assets, coded dialogs, custom adapters, middleware or custom actions. They are just regular NuGet or npm packages (depending on the code language for your bot). You'll use the Bot Framework CLI tool to merge the package's declarative contents with your bot (package management in Composer will handle this for you, or you could do it yourself using the Bot Framework CLI tool). They can be made up of any combination of declarative assets (.dialog, .lu, .lg, .qna files), coded extensions (custom actions, middleware, adapters), or just plain old package libraries.
```
Comment (peterinnesmsft) [inline docs/overview.md:21]: I would remove middleware for now and stick to custom actions and adapters. We still don't have a solidified story for middleware.

File: docs/extending-with-code.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# Extending your bot with code
+
+> Note: You should be familiar with the existing documentation on [custom actions](https://docs.microsoft.com/en-us/composer/how-to-add-custom-action) and [declarative dialogs](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-dialogs-declarative?view=azure-bot-service-4.0). This page is supplemental to those existing resources.
+
+Extending a bot built on the new Adaptive Runtime with code is slightly different than with the current runtime available in Composer. The Adaptive Runtime is designed to be treated as a black-box dependency that can dynamically register your components, so you'll never modify it directly. Instead, you'll use the new `IPlugin` interface to let the runtime know at compile-time there are new components to include.
```
Comment (peterinnesmsft) [inline docs/extending-with-code.md:5]: IBotPlugin

**Review by peterinnesmsft — APPROVED** [review]

**Pull request merged**